src/test/java/uk/gov/hmcts/reform/wastandalonetaskbpmn/bpmn/CamundaCreateTaskTest.java

```diff
@@ -18,4 +27,76 @@ public void should_create_a_task_with_no_delay_until() {
         createTask(false);
     }
 
+    @Test
+    public void should_not_create_a_task_with_different_TenantId_multiple_resource() {
+
+        clearDeployments();
+
+        RepositoryService repositoryService = processEngineRule.getRepositoryService();
+
+        DeploymentBuilder deploymentBuilder = repositoryService.createDeployment()
+            .tenantId("someTenantId");
+        deploymentBuilder.addClasspathResource("wa-task-initiation-ia-asylum.bpmn")
+            .deploy();
+
+        DeploymentBuilder deploymentBuilder2 = repositoryService.createDeployment()
+            .tenantId("someTenantId2");
+
+        deploymentBuilder2.addClasspathResource("wa-task-initiation-ia-asylum_duplicate.bpmn")
+            .deploy();
+
+        assertThrows(MismatchingMessageCorrelationException.class, () -> {
+            createTask(false);
```

Comment:
I assume this create does not specify a tenantId and therefore does not correlate to any process?  Is this correct?

Reply:
Yes Ravi. Create task method is configured to look for bpmn's with no tenantId.